Panic in light client db

Reported from the cosmos discord channel for IBC
```
panic: can't open light client database: resource temporarily unavailable

goroutine 411600 [running]:
github.com/cosmos/relayer/relayer.(Chain).MustGetLatestLightHeight(...)
        /home/microtick/relayer/relayer/tm-light-client.go:269
github.com/cosmos/relayer/relayer.(relayMsgRecvPacket).FetchCommitResponse.func1(0x1481000, 0x1)
        /home/microtick/relayer/relayer/relayPackets.go:137 +0x265
github.com/avast/retry-go.Do(0xc00006ee98, 0xc00006ee78, 0x4, 0x4, 0x97, 0x0)
        /home/microtick/go/pkg/mod/github.com/avast/retry-go@v2.6.0+incompatible/retry.go:114 +0x15f
github.com/cosmos/relayer/relayer.(relayMsgRecvPacket).FetchCommitResponse(0xc003114ec0, 0xc000cfdd40, 0xc000cfdc20, 0x0, 0x0)
        /home/microtick/relayer/relayer/relayPackets.go:135 +0x145
github.com/cosmos/relayer/relayer.(NaiveStrategy).sendTxFromEventPackets(0xc000e7f600, 0xc000cfdd40, 0xc000cfdc20, 0xc0054ce880, 0x1, 0x1)
        /home/microtick/relayer/relayer/naive-strategy.go:345 +0x99
github.com/cosmos/relayer/relayer.(*NaiveStrategy).HandleEvents(0xc000e7f600, 0xc000cfdd40, 0xc000cfdc20, 0xc002ffa6c0)
        /home/microtick/relayer/relayer/naive-strategy.go:237 +0x85
created by github.com/cosmos/relayer/relayer.relayerListenLoop
        /home/microtick/relayer/relayer/strategies.go:138 +0xcd1
Disk is not full, and I tried some questionable things like deleting .relayer/light/chain_id.db/LOCK without success
```

--- a/relayer/tm-light-client.go
+++ b/relayer/tm-light-client.go
@@ -7,6 +7,7 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"sync"
 	"time"
 
 	retry "github.com/avast/retry-go"
@@ -23,11 +24,17 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
-// NOTE: currently we are discarding the very noisy light client logs
-// it would be nice if we could add a setting the chain or otherwise
-// that allowed users to enable light client logging. (maybe as a hidden prop
-// on the Chain struct that users could pass in the config??)
-var logger = light.Logger(log.NewTMLogger(log.NewSyncWriter(ioutil.Discard)))
+var (
+	// NOTE: currently we are discarding the very noisy light client logs
+	// it would be nice if we could add a setting the chain or otherwise
+	// that allowed users to enable light client logging. (maybe as a hidden prop
+	// on the Chain struct that users could pass in the config??)
+	logger = light.Logger(log.NewTMLogger(log.NewSyncWriter(ioutil.Discard)))
+
+	// a lock to prevent two processes from trying to access the light client
+	// database at the same time resulting in errors and panics.
+	lightDBMutex sync.Mutex
+)
 
 func lightError(err error) error { return fmt.Errorf("light client: %w", err) }
 
@@ -171,18 +178,19 @@ func (c *Chain) LightClient(db dbm.DB) (*light.Client, error) {
 // NewLightDB returns a new instance of the lightclient database connection
 // CONTRACT: must close the database connection when done with it (defer df())
 func (c *Chain) NewLightDB() (db *dbm.GoLevelDB, df func(), err error) {
-	if err := retry.Do(func() error {
-		db, err = dbm.NewGoLevelDB(c.ChainID, lightDir(c.HomePath))
-		if err != nil {
-			return fmt.Errorf("can't open light client database: %w", err)
-		}
-		return nil
-	}, rtyAtt, rtyDel, rtyErr); err != nil {
-		return nil, nil, err
+	// a lock is used to prevent error messages or panics from two processes
+	// trying to simultanenously use the light client
+	lightDBMutex.Lock()
+
+	db, err = dbm.NewGoLevelDB(c.ChainID, lightDir(c.HomePath))
+	if err != nil {
+		lightDBMutex.Unlock()
+		return nil, nil, fmt.Errorf("can't open light client database: %w", err)
 	}
 
 	df = func() {
 		err := db.Close()
+		lightDBMutex.Unlock()
 		if err != nil {
 			panic(err)
 		}

